splunk_handler/__init__.py

```diff
@@ -31,6 +21,17 @@ def __init__(self, host, port, username, password, index):
         self.username = username
         self.password = password
         self.index = index
+        self.source = source
+        self.sourcetype = sourcetype
+
+        if hostname is None:
+            self.hostname = socket.gethostname()
```

Comment:
@sullivanmatt this will actually work a lot better since the hostname was being redetermined on every log call before.